core/src/main/resources/reference.conf

```diff
@@ -541,3 +541,126 @@ jdbc-read-journal {
     }
   }
 }
+
+# the akka-persistence-durable-state-store in use
+jdbc-durable-state-store {
+  class = "akka.persistence.jdbc.state.scaladsl.DurableStateStore"
+
+  # number of records fetched from the store at once
+  batchSize = 500
+  # New states are retrieved (polled) with this interval.
+  refreshInterval = "1s"
+  # can be H2, POSTGRES
+  # schemaType = H2
+  schemaType = Postgres
```

Comment:
Is that so that we will only support it for Postgres and H2? Meaning, only Postgres (for prod)? 

> should it have a default value? does the journal have a default value for this?

I don't think so. Specially if we don't want/plan to support other DBs.

Reply:
@patriknw @octonato The `schemaType` needs to have a value since we have some platform specific code in the implementation. The statement that fetches the next value of the sequence differs between Postgres and H2 (haven't yet looked at other implementations though). Hence when building the sequence updater I have code like this ..

```scala
lazy val sequenceNextValUpdater = durableStateTableCfg.schemaType match {
  case "H2"       => new H2SequenceNextValUpdater(profile)
  case "Postgres" => new PostgresSequenceNextValUpdater(profile)
}
```

Hence the config needs to have the `schemaType` with a default value. This can of course be changed by changing the config when required (e.g. to run tests).

Reply:
The user should only choose database at one place, which is the Slick profile: `akka-persistence-jdbc.shared-databases.slick.profile`. We should be able to look at the profile choice and pick the right SequenceNextValUpdater implementation from that.

Reply:
@patriknw Now fixed. `schemaType` removed from `reference.conf`